[sarang_daddy] News_Stand 1주차

sarangdaddy opened the pull request:
## 학습 계획

- [x] 월요일
  - [x] 키워드 학습
  - [x] Grid 학습
  - [x] Feature List 작성
- [x] 화요일
  - [x] DATA 획득 생각하기
  - [x] 계획서 작성
- [x] 수요일
  - [x] db.json 만들기
  - [x] 기능 구현 시작
 - [ ] 목요일
   - [x] 헤더 데이터 가져오기
   - [x] 헤드라인 데이터 가져오기
   - [ ] 언론사 데이터 가져오기
 - [ ] 금요일
   - [ ] 헤드라인 롤링 UX
   - [ ] 언론사 캐러셀 

## 고민사항
- ~~Feature List를 먼저 작성하고 세부 학습 계획을 세우고자 한다.~~
- ~~DATA를 어떻게 받아올지 고민~~
- ~~DATA 획득을 먼저 생각해야 설계서 작성이 가능 할 것 같다.~~
- 롤링 기능 구현 고민... 

## 학습자료
[학습](https://stitch-dart-ccd.notion.site/FE-5-c53fa71f543d4bc89314ace45c585241)
[설계](https://stitch-dart-ccd.notion.site/67df44200d0e4a6c977c39c6bd960244?v=4c93fbcb82234ceb8ca7a70212140d92)

sarangdaddy committed:
[Docs] 뉴스 스탠드 1주차 학습계획 업데이트

- README 파일 생성

sarangdaddy committed:
[set] 기초 폴더 셋팅

- html, css, src 폴더 생성
- 사용 svg 파일 추가
- html, style.css 추가

sarangdaddy committed:
[Addfile] db.json 파일 추가

- headLine DATA 가져와서 업데이트

ahnlook commented on css/common/fonts.css:
```diff
@@ -0,0 +1,7 @@
+@font-face {
+  src: url("https://cdn.jsdelivr.net/gh/orioncactus/pretendard@v1.3.6/dist/web/static/pretendard.css");
+  font-family: -apple-system, BlinkMacSystemFont, "Apple SD Gothic Neo",
+    "Pretendard Variable", Pretendard, Roboto, "Noto Sans KR", "Segoe UI",
```
요구사항에 Pretendard를 적용하라고 되어 있는데 font-family 순서가 중간인 특별한 이유가 있을까요?

sarangdaddy replied:
시스템에 가능한 맞추고자 한다면 추천받은 구성입니다.
https://github.com/orioncactus/pretendard

ahnlook commented on css/styles.css:
```diff
@@ -0,0 +1,15 @@
+
+/* SCREENS */
+
+h1 {
```
`h1` 태그를 많이 사용하셔서 따로 속성 값을 주신 건가요?

sarangdaddy replied:
폰트 적용이 잘 됬는지 확인해보려고 적었어요~ 혼동을 주니 지워야겠어요 ^^;

ahnlook reviewed (commented):
요일별 계획이 되어 있어서 하루 하루 뭘 해야 되는지에 대한 고민이 줄어 하루를 효율적으로 보낼 수 있을 것 같요! :)

sarangdaddy committed:
[Addfile] db.json에 journalList 데이터 추가

sarangdaddy committed:
[Docs] README update

sarangdaddy committed:
[Design] 기본 UI 선정 중

- HTML과 CSS로 기본 배치 및 UI 요소 정리
- 추후 HTML 정보는 JS로 부여 예정

sarangdaddy committed:
[Design] HTML, CSS 레이아웃 구성 완료

- 구성을 토대로 JS에서 HTML 구조 만들어야함

sarangdaddy committed:
[Rename] app.js -> main.js 로파일명 변경

- 전체 어플리케이션이 아닌 news-stand를 실행하는 개념으로 이름명 변경
- HTML - scipt : type='module' 추가

sarangdaddy committed:
[Feat] 뉴스스탠드 header JS 동적 기능 추가

- JS로 news-stand header 구현

sarangdaddy committed:
[Feat] headLine 데이터 가져오기

- fetch API를 통해 db.json에서 데이터 불러오기 성공

sarangdaddy committed:
[Feat] 헤드라인 데이터 db.json에서 가져오기

- JS와 fetch API를 사용하여 동적으로 데이터 가져오기 완료

sarangdaddy committed:
[Feat] 언론사 영역 추가

- 언론사 영역 엘리먼트 추가 완료
- 언론사 헤더 추가 완료
- 언론사 db 연결 완료

sarangdaddy committed:
[Feat] 언론사 리스트 96개 가져오기

- 1. main에서 journalListView에게 언론사 리스트 요청
- 2. journalListView에서 getJournalListData에게 URL을 전송해서 DATA 요청
- 3. getJournalListData에서 fetch API를 통해 DATA를 요청하고
- 4. 응답받은 데이터를 가진 journal 생성자 생성
- 5. journal생성자는 journal EL를 만들어준다.

sarangdaddy committed:
[Style] console.log 제거

sarangdaddy committed:
[Design] 언론사 로고 사이즈 수정

- grid셀의70%로 조정

sarangdaddy committed:
[Design] 언론사 캐러셀을 위한 CSS 변경

- 캐러셀을 위한 영역 div class="journal-carousel" 추가
- 최초 journal-container 배치 이동 left:-900px
- journal-carousel 오버플로우 히든 적용

sarangdaddy committed:
[Feat] 언론사 리스트 순서 랜덤으로 부여

- DB에서 받아온 언론사 리스트를 랜덤으로 솔트 후 appendChild 진행

crongro commented on src/api/getHeadLineData.js:
```diff
@@ -0,0 +1,11 @@
+import { HeadLine } from "../components/headLine/headLineClass.js";
+
+export const getHeadLine = (URL) => {
```
이렇게 파일 한개에 함수 한개를 표현하면 너무 많은 파일로 관리가 어려울것 같아요.
data 요청하는 함수를 한개 파일로 모아두어도 좋을것 같아요.

crongro commented on src/api/getJournalListData.js:
```diff
@@ -0,0 +1,11 @@
+import { Journal } from "../components/journalList/journalClass.js";
+
+export const getJournal = (URL) => {
+  return fetch(URL)
+    .then((response) => response.json())
+    .then((jsonData) => {
+      const journal = new Journal(jsonData);
+      return journal.makeJournal();
+    })
+    .catch((error) => console.error(`fetch 에러! ${error}`));
```
then 메서드가 익숙해지면 async 함수도 활용해보시고요.

crongro commented on src/components/headLine/headLineClass.js:
```diff
@@ -0,0 +1,24 @@
+    this.headLineList = jsonData;
+  }
+
+  toBeElement = function (title) {
```
클래스안에 메서드는 function 키워드 삭제 가능하죠.

toBeElement(title) {
}

crongro commented on src/components/headLine/headLineClass.js:
```diff
@@ -0,0 +1,24 @@
+  };
+
+  makeHeadline = function () {
+    // this.headLineList.forEach((title) => {
+    //   this.toBeElement(title.headLineList);
+    // });
```
커밋할때 주석은 늘 삭제~

crongro commented on src/components/journalList/journalClass.js:
```diff
@@ -0,0 +1,16 @@
+  }
+
+  makeJournal() {
+    return this.journalList.reduce((acc, cur) => {
```
reduce 활용도 좋긴한데, 지금은 map을 써도 되는거네요.
조금더 간단하게 표현될 듯 합니다.

crongro commented on src/getData/journalData.js:
```diff
@@ -0,0 +1,19 @@
+// 네이버 콘솔에서 데이터를 가져오는 코드
+
+const result = [];
+let id = 1;
+
+const journalList = document.querySelectorAll(".thumb img");
+
+journalList.forEach((listEl) => {
+  const obj = {
+    journalId: id,
+    journalAlt: listEl.alt,
+    journalSrc: listEl.currentSrc,
+  };
+
+  result.push(obj);
+  id++;
+});
+
+console.log(result);
```
데이터조작은 함수로 감싸서 하시면 좋겠어요~

crongro reviewed (approved):
수고했어요. 여러 모듈로 나눈 시도가 좋고요.
template literal 을 적극 더 활용해보세요.
그리고, **모든 로직은 함수나 클래스 안에 메서드로 옮겨서 표현하시면 더 좋을것 같네요.**
